### PR TITLE
syslog-ng-otlp()

### PR DESCRIPTION
**Pull request by zsoltgyulai94**
Documented syslog-ng-otlp() source and destination.

Signed-off-by: Zsolt Gyulai (zgyulai) zsolt.gyulai@quest.com

### CODE REVIEW

File: doc/_admin-guide/070_Destinations/315_syslog-otlp/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+* An integrated application layer acknowledgement is available.
+* Google service authentication (ADC or ALTS), and improved load balancing are supported.
+
+### Example: Configure syslog-ng-otlp() destination on the sender node
+
+```config
+destination d_syslog_ng_otlp {
+  syslog-ng-otlp(url("your-receiver-syslog-ng-instance:4317"));
+};
+```
```
Comment (HofiOne) [inline doc/_admin-guide/070_Destinations/315_syslog-otlp/README.md:20]: the source example would be useful here showing a simple pair of syslog-ng instances using otlp

the source example could get a short title that mentions `your-receiver-syslog-ng-instance` as the other pair

File: doc/_admin-guide/060_Sources/175_syslog-otlp/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+* An integrated application layer acknowledgement is available.
+* Google service authentication (ADC or ALTS), and improved load balancing are supported.
+
+### Example: Configure syslog-ng-otlp() source on the receiver node
+
+```config
+source s_syslog_ng_otlp {
+  syslog-ng-otlp(port(4317));
+};
+```
```
Comment (HofiOne) [inline doc/_admin-guide/060_Sources/175_syslog-otlp/README.md:20]: just like for the destination, maybe it is worth mentioning proper pairs here

File: doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+*Description:* This option can be used to enable or disable the chained hostname format. For more information, see the chain-hostnames() global option.
+
+{% include doc/admin-guide/options/channel-args.md %}
+
+## concurrent-requests()
+
+|   Type:|       integer|
+|Default:|             2|
+
+*Description:* This option configures the upper limit of in-flight gRPC requests per worker. It is advisd to set this value in the range of 10s or 100s when there are a high number of clients sending simultaneously. In an optimzed solution, the number of `workers()` and `concurrent-requests()` is greater than or equal to the number of clients. However, this can cause an increase in memory usage.
```
Comment (HofiOne) [inline doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md:81]: tiny typo `optimzed`

File: doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+*Description:* This option assigns a facility value to messages received from the file source if the message does not specify one.
+
+## default-level()
+
+|   Type:|       string|
+|Default:|       notice|
+
+*Description:* This option defines the default level value if the `PRIORITY` entry does not exist.
+
+## default-priority()
```
Comment (HofiOne) [inline doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md:97]: this kind of options might be worth having separate snippets as well

File: doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+|Default:|              `no`|
+
+*Description:* This option enables or disables the DNS cache usage.
+
+## ebpf()
+
+Available in syslog-ng OSE 4.2 and later versions.
+
+If this option is not used, the kernel chooses the receive socket for a specific UDP randomly based on the source IP/port of the sender. It is possible to customize this algorithm using the Extended Berkeley Packet Filter (eBPF) plugin. The `ebpf()` option changes the `SO_REUSEPORT` algorithm of the kernel, to place messages randomly into one of the UDP sockets. The decision which UDP socket buffer a datagram is placed is made for every datagram, and not once for every stream. This results in the percet load-balancing of messages across the set of UDP sockets. While this resolves the imbalance between the sockets and results in perfect load balancing, the order of messages from the same sender is lost, which is the drawback of increased throughput.
+
```
Comment (HofiOne) [inline doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md:116]: en example config would be useful here I guess

File: doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+## flags()
+
```
Comment (HofiOne) [inline doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md:118]: this flag() option is surely worth being moved to a dedicated snippet I guess

**Review by HofiOne — CHANGES_REQUESTED** [review]

File: doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md (hunk 5 above)
Comment (zsoltgyulai94) [reply doc/_admin-guide/060_Sources/175_syslog-otlp/000_otlp_source_options.md:111]: Added one based on [this](https://github.com/syslog-ng/syslog-ng/pull/4365).
Let me know if it's not quite right.

**Review by HofiOne — APPROVED** [review]

**Pull request merged**